CHANGELOG.md

```diff
@@ -4,6 +4,243 @@ All notable changes to Router will be documented in this file.
+                access_key_id: "my-access-key"
+                secret_access_key: "my-secret-access-key"
+                region: "us-east-1"
+                service_name: "vpc-lattice-svcs" # "s3", "lambda" etc.
+```
+
+The full documentation can be found in the [router documentation](https://www.apollographql.com/docs/router/configuration/authn-subgraph).
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) and [@BlenderDude](https://github.com/BlenderDude) in https://github.com/apollographql/router/pull/3365
+
+### Helm: add init containers to deployemt ([Issue #3248](https://github.com/apollographql/router/issues/3248))
+
+This is a new option when deploying the router, so that containers may be specified which execute before the Router container (or any extra Containers) begin executing. You can read more about [init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) in the kubernetes documentation.
+
+By [@laszlorostas](https://github.com/laszlorostas) in https://github.com/apollographql/router/pull/3444
+
+### Helm: exposes the lifecycle object on the router container ([Issue #3563](https://github.com/apollographql/router/issues/3563))
+
+You can now set the [kubernetes lifecycle object](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/) on the router container in the helm chart.
+
+By [@bjoernw](https://github.com/bjoernw) in https://github.com/apollographql/router/pull/3570
+
+## 🐛 Fixes
+
+### Require the main (GraphQL) route to shutdown before other routes ([Issue #3521](https://github.com/apollographql/router/issues/3521))
+
+Router shutdown sequence has been improved to ensure that the main (GraphQL) route is shutdown before other routes are shutdown. Prior to this change all routes shut down in parallel and this would mean that, for example, health checks stopped responding prematurely.
+
+This was particularly undesirable when the router is executing in Kubernetes, since continuing to report live/ready checks during shutdown is a requirement.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3557
+
+### spelling of content_negociation corrected to content_negotiation ([Issue #3204](https://github.com/apollographql/router/issues/3204))
+
+spelling mistake in content_negociation is fixed
+
+By [@krishna15898](https://github.com/krishna15898) in https://github.com/apollographql/router/pull/3162
+
+### Fix redis reconnections ([Issue #3045](https://github.com/apollographql/router/issues/3045))
+
+Redis reconnection policy was using an exponential backoff delay with a maximum number of attempts. Once that maximum is reached, reconnection was never tried again (there's no baseline retry). The router will now always retry with a maximum delay of 2 seconds, and a timeout of 1 millisecond on redis commands, so that the router can continue serving requests in the meantime.
```

Comment:
```suggestion
Redis reconnection policy was using an exponential backoff delay with a maximum number of attempts. Once that maximum is reached, reconnection was never tried again (there's no baseline retry). The router will now always retry with a maximum delay of 2 seconds, and a timeout of 1 millisecond, so that the router can continue serving requests in the meantime.
```